packages/playwright/src/common/config.ts

```diff
@@ -206,6 +206,15 @@ function pathResolve(baseDir: string, relative: string | undefined): string | un
   return path.resolve(baseDir, relative);
 }
 
+function handleMaxFailures(maxFailures: number | string): number {
+  if (typeof maxFailures === 'string') {
+    const parsedMaxFailures = parseInt(maxFailures, 10);
+    return maxFailures.endsWith('%') ? Math.min(0.99, parsedMaxFailures / 100) : parsedMaxFailures;
```

Comment:
Here I had to set 0.99 because if I specify 1 I will not be able to correctly calculate the value relative to the total number of tests.  I mean that I will not be able to distinguish between the value `1` (static) and `100%` (dynamic, which will be transformed to `1`).

Reply:
You should pass a number | string instead of this, or some up with a data structure. 0.99 is not great.